pkg/util/watchdog/watchdog_test.go

```diff
@@ -215,8 +215,11 @@ var _ = Describe("SimpleWatchdog", func() {
 			watchdog.Start(ctx)
 			close(doneCh)
 		}()
-		Expect(watchdog.HasTicked(false)).Should(BeFalse())
-		Consistently(hasTicked).ShouldNot(Receive())
+		// first onTick call
```

Comment:
maybe you can pass function to OnTick that blocks on something, i.e
```go
OnTick: func(ctx context.Context) error {
    <-onTickBlockCh
    return nil
},
```
and then you can check
```go
// before the first OnTick returns false
Expect(watchdog.HasTicked(false)).Should(BeFalse())
// unblock OnTick
close(onTickBlockCh)
// after the first tick returns true
Expect(watchdog.HasTicked(false)).Should(BeTrue())
Expect(hasTicked).Should(BeClosed())
```